src/Utility/Text.php

```diff
@@ -236,7 +236,7 @@ public static function insert(string $str, array $data, array $options = []): st
         /** @var array<string, mixed> $dataReplacements */
         $dataReplacements = array_combine($hashKeys, array_values($data));
         foreach ($dataReplacements as $tmpHash => $tmpValue) {
-            $tmpValue = is_array($tmpValue) ? '' : $tmpValue;
+            $tmpValue = is_string($tmpValue) ? $tmpValue  : '';
```

Comment:
I dont think that fix is correct
If by any means an integer comes out of the loop, you don't want this to be an empty string.

You should instead use casting to solve the issue.

Please also provide a test case to back up the issue.